0.data_analysis/plate_correlation_analyses/nbconverted/well_agg_plate_genotype_correlation_analysis.py

```diff
@@ -0,0 +1,170 @@
+
+# # Well-Aggregated Plate and Genotype Correlation Analysis
+# Correlations between groups defined by genotype and plate are determined to understand the similarities between group morphologies.
+# These correlations are computed between cell morphologies aggregated to the well level after feature selection.
+
+# In[1]:
+
+
+import pathlib
+import sys
+
+import pandas as pd
+
+# Path to correlation class
+sys.path.append(
+    "../utils"
+)
+
+# Class for calculating correlations
+from CorrelateData import CorrelateData
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# # Inputs
+
+# In[3]:
+
+
+data_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles").resolve(strict=True)
+
+plate3df_path = pathlib.Path(root_dir / data_path / "Plate_3_bulk_camerons_method.parquet").resolve(strict=True)
+plate3pdf_path = pathlib.Path(root_dir / data_path / "Plate_3_prime_bulk_camerons_method.parquet").resolve(strict=True)
+plate5df_path = pathlib.Path(root_dir / data_path / "Plate_5_bulk_camerons_method.parquet").resolve(strict=True)
+
+plate3df = pd.read_parquet(plate3df_path)
+plate3pdf = pd.read_parquet(plate3pdf_path)
+plate5df = pd.read_parquet(plate5df_path)
+
+
+# # Outputs
+
+# In[4]:
+
+
+plate_correlation_path = pathlib.Path("construct_correlation_data/well_agg_plate_genotype_correlations.parquet")
+plate_correlation_path.mkdir(parents=True, exist_ok=True)
+
+
+# # Process Bulk Plate Data
+
+# ## Combine data
+# Concat plate data and retain common columns.
+
+# In[5]:
+
+
+plates_cols = plate3df.columns.intersection(plate3pdf.columns).intersection(plate5df.columns)
+platesdf = pd.concat([plate3df, plate3pdf, plate5df], axis=0)
+platesdf = platesdf[plates_cols]
+
+
+# In[6]:
+
+
+# Morphology and metadata columns
+morph_cols = [col for col in platesdf.columns if "Metadata" not in col]
+meta_cols = platesdf.columns.difference(morph_cols)
+
+
+# # Correlate wells
+# Wells are correlated between plate and genotype.
+
+# In[7]:
+
+
+cd = CorrelateData()
+correlationsdf = []
+
+
+# ## Well Correlations (same genotypes different plates)
+
+# In[8]:
+
+
+for genotype in platesdf["Metadata_genotype"].unique():
+
+    correlation_params = {
+    }
+
+    correlationsdf.append(
+        cd.inter_correlations(
+            _df=platesdf.loc[platesdf["Metadata_genotype"] == genotype].copy(),
+            _antehoc_group_cols=["Metadata_Plate"],
+            _feat_cols=morph_cols,
+            _posthoc_group_cols=["Metadata_Well", "Metadata_genotype"],
+            _drop_cols=["Metadata_Well"]
```

Comment:
I recommend putting small notes on each parameter in this notebook. Like I am a bit confused as to why there is a parameter called `_drop_cols` and why Well would be dropped. I also think the terms `antehoc` and `posthoc` in this context aren't as interpretable as they could be. Feel free to also provide clarity here as well.

Reply:
Here, I was trying to stay more DRY by not re-describing the documentation in the correlation util. This code is also necessarily more abstract, since this code is capable of computing correlations between groups in projects outside the scope of the NF1 project.

Abstract code can sometimes be difficult to grasp, so I proposed an enhancement (see #59), which I think is a lower priority for now. In some cases, such as this one, I think it's more beneficial to view the lower-level code and corresponding documentation to understand the finer details. There should also be a level 2 markdown heading above each of these function calls describing the groups being correlated using similar terminology to the column names passed as parameter values.

Also, the `_drop_cols` is mentioned in the correlation util documentation, and only determines which column(s) are excluded from the returned output. However, it's optional.